File: leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/477-total-hamming-distance.js
var totalHammingDistance = function(nums) {
    let total = 0;
    const n = nums.length;
    for (let i = 0; i < 32; i++) {
        let count = 0;
        for (const num of nums) {
            count += (num >> i) & 1;
        }
        total += count * (n - count);
    }
    return total;
};

File: leetcode/AIcoding/3.DeepSeek-v3/rust_solutions/477-total-hamming-distance.rs
impl Solution {
    pub fn total_hamming_distance(nums: Vec<i32>) -> i32 {
        let mut total = 0;
        let n = nums.len() as i32;
        for i in 0..32 {
            let mut count = 0;
            for &num in &nums {
                count += (num >> i) & 1;
            }
            total += count * (n - count);
        }
        total
    }
}